platform/cosmos/api.go

```diff
@@ -41,86 +41,98 @@ func (p *Platform) CurrentBlockNumber() (int64, error) {
 }
 
 func (p *Platform) GetTxsByAddress(address string) (blockatlas.TxPage, error) {
-	srcTxes := make([]Tx, 0)
-
-	tagsList := []string{"recipient", "sender", "delegator", "destination-validator"}
-
-	for _, tag := range tagsList {
-		responseTxes, _ := p.client.GetAddrTxes(address, tag)
-		srcTxes = append(srcTxes, responseTxes...)
-	}
-
-	normalisedTxs := make([]blockatlas.Tx, 0)
-
-	for _, srcTx := range srcTxes {
-		normalisedInputTx, ok := Normalize(&srcTx)
-		if ok {
-			normalisedTxs = append(normalisedTxs, normalisedInputTx)
-		}
+	srcTxs := make([]Tx, 0)
+	tagsList := []string{"transfer.recipient", "message.sender"}
+
+	var wg sync.WaitGroup
+	for _, t := range tagsList {
+		wg.Add(1)
+		go func(tag, addr string) {
+			defer wg.Done()
+			txs, _ := p.client.GetAddrTxs(addr, tag)
+			srcTxs = append(srcTxs, txs.Txs...)
+		}(t, address)
 	}
-
-	sort.Slice(normalisedTxs, func(i, j int) bool {
-		return normalisedTxs[i].Date > normalisedTxs[j].Date
-	})
-
-	return normalisedTxs, nil
+	wg.Wait()
+	return NormalizeTxs(srcTxs), nil
 }
 
 // NormalizeTxs converts multiple Cosmos transactions
-func NormalizeTxs(srcTxs []Tx, pageSize int) (txs []blockatlas.Tx) {
+func NormalizeTxs(srcTxs []Tx) blockatlas.TxPage {
+	txMap := make(map[string]bool)
+	txs := make(blockatlas.TxPage, 0)
 	for _, srcTx := range srcTxs {
-		tx, ok := Normalize(&srcTx)
-		if !ok || len(txs) >= pageSize {
+		_, ok := txMap[srcTx.ID]
+		if ok {
 			continue
 		}
-		txs = append(txs, tx)
+		normalisedInputTx, ok := Normalize(&srcTx)
+		if ok {
+			txMap[srcTx.ID] = true
+			txs = append(txs, normalisedInputTx)
+		}
 	}
-	return
+	return txs
 }
 
 // Normalize converts an Cosmos transaction into the generic model
 func Normalize(srcTx *Tx) (tx blockatlas.Tx, ok bool) {
-	date, _ := time.Parse("2006-01-02T15:04:05Z", srcTx.Date)
-	block, _ := strconv.ParseUint(srcTx.Block, 10, 64)
+	date, err := time.Parse("2006-01-02T15:04:05Z", srcTx.Date)
+	if err != nil {
+		return blockatlas.Tx{}, false
+	}
+	block, err := strconv.ParseUint(srcTx.Block, 10, 64)
+	if err != nil {
+		return blockatlas.Tx{}, false
+	}
 	// Sometimes fees can be null objects (in the case of no fees e.g. F044F91441C460EDCD90E0063A65356676B7B20684D94C731CF4FAB204035B41)
-	var fee string
-	if len(srcTx.Data.Contents.Fee.FeeAmount) == 0 {
-		fee = "0"
-	} else {
-		fee, _ = util.DecimalToSatoshis(srcTx.Data.Contents.Fee.FeeAmount[0].Quantity)
+	fee := "0"
+	if len(srcTx.Data.Contents.Fee.FeeAmount) > 0 {
+		fee, err = util.DecimalToSatoshis(srcTx.Data.Contents.Fee.FeeAmount[0].Quantity)
+		if err != nil {
+			return blockatlas.Tx{}, false
+		}
 	}
 
 	tx = blockatlas.Tx{
-		ID:    srcTx.ID,
-		Coin:  coin.ATOM,
-		Date:  date.Unix(),
-		Fee:   blockatlas.Amount(fee),
-		Block: block,
-		Memo:  srcTx.Data.Contents.Memo,
+		ID:     srcTx.ID,
+		Coin:   coin.ATOM,
+		Date:   date.Unix(),
+		Status: blockatlas.StatusCompleted,
```

Comment:
I think it could be an error as well

https://www.mintscan.io/txs/5e78c65a8c1a6c8239ebbbbf2e42020e6adba8037edea83bf88e1a9159cf13b8

can you check how to pull status from this errored transaction?